pkg/comp-functions/functions/spks/spksmariadb/tls.go

```diff
@@ -0,0 +1,120 @@
+package spksmariadb
+
+import (
+	"context"
+	"fmt"
+
+	v1 "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
+	xfnproto "github.com/crossplane/function-sdk-go/proto/v1"
+	xhelm "github.com/vshn/appcat/v4/apis/helm/release/v1beta1"
+	spksv1alpha1 "github.com/vshn/appcat/v4/apis/syntools/v1alpha1"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/functions/common"
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// HandleTLS will enable TLS if it's specified in the composite.
+// It will then deploy certmanager objects to generate certificates.
+func HandleTLS(ctx context.Context, comp *spksv1alpha1.CompositeMariaDBInstance, svc *runtime.ServiceRuntime) *xfnproto.Result {
+	err := svc.GetObservedComposite(comp)
+	if err != nil {
+		return runtime.NewFatalResult(fmt.Errorf("cannot get composite for spks redis: %w", err))
+	}
+
+	if !comp.Spec.Parameters.TLS {
```

Comment:
API comment says "RequireTLS: Enforce TLS on this instance. Needs tls to be true" but there's no validation. If someone sets `RequireTLS: true` with `TLS: false`, the behavior is undefined.